server/controllers/finance/cash.js

```diff
@@ -1,77 +1,54 @@
-
 /**
  * Cash Controller
  *
- *
  * This controller is responsible for processing cash payments for patients. The
- * payments can either be against an previous invoice (invoice payment) or a future
- * invoice (cautionary payment).
+ * payments can either be against an previous invoice (invoice payment) or a
+ * future invoice (cautionary payment).
  *
  * In order to reduce the burden of accounting on the user, the user will first
  * select a cashbox which implicitly bundles in cash accounts for all supported
  * currencies.  The API accepts a cashbox ID during cash payment creation and
  * looks up the correct account based on the cashbox_id + currency.
- *
  * @module finance/cash
  *
- * @requires node-uuid
  * @requires lib/db
- * @requires cash.create
+ * @requires lib/filters
+ * @requires lib/barcode
  * @requires lib/errors/NotFound
  * @requires lib/errors/BadRequest
+ * @requires config/identifiers
+ * @requires cash.create
  */
 
 const _ = require('lodash');
 
 const db = require('../../lib/db');
-
-const NotFound = require('../../lib/errors/NotFound');
-const BadRequest = require('../../lib/errors/BadRequest');
-
-const identifiers = require('../../config/identifiers');
 const barcode = require('../../lib/barcode');
-
 const FilterParser = require('../../lib/filter');
-
+const { BadRequest, NotFound } = require('../../lib/errors');
+const identifiers = require('../../config/identifiers');
 const cashCreate = require('./cash.create');
 
-const entityIdentifier = identifiers.CASH_PAYMENT.key;
-
-/** retrieves the details of a cash payment */
 exports.detail = detail;
-
-/** retrieves a list of all cash payments */
-/** search cash payment by filtering */
 exports.read = read;
-
-/** creates cash payments */
 exports.create = cashCreate;
-
-/** modifies previous cash payments */
 exports.update = update;
-
-/** searches for a cash payment's uuid by their human-readable reference */
-exports.reference = reference;
-
-/** lookup a cash payment by it's uuid */
 exports.lookup = lookup;
-
-/** list all cash payment */
-exports.listPayment = listPayment;
-
-/** checkInvoicePayment if the invoice is paid */
+exports.find = find;
 exports.checkInvoicePayment = checkInvoicePayment;
+exports.safelyDeleteCashPayment = safelyDeleteCashPayment;
```

Comment:
This function should not be exposed since it is deleting a record without checking the opportunity of doing that. It should be there for internal use.

Reply:
Where do you suggest it should go?

Reply:
Expose an other function which is doing checking first and then call the ```safelyDeleteCashPayment```.

Reply:
What is about this issue ?